client/galaxy/scripts/apps/login.js

```diff
@@ -0,0 +1,39 @@
+
+var jQuery = require( 'jquery' ),
+    $ = jQuery,
+    PANEL = require( 'layout/panel' ),
+    _l = require( 'utils/localization' ),
+    PAGE = require( 'layout/page' );
+
+window.app = function app( options, bootstrapped ){
```

Comment:
> Why do we need globals here i.e. what if we want to load multiple apps into one window.

Well, we need the global app variable to have something we can pass the options and bootstrapped data in the js-app.mako file.

As for multiple apps in one window - the general idea in web application development is to have one app per window. Eventually, we'd boil galaxy into a single app/server-route web application, but for now our next steps should be: 
* replacing server side routes, controller methods, and mako files with Backbone.router, models, and views
* building apps from the views above for a select group of functionality (workflows, Pages, etc.), 
* filling in needed API functionality as we replace post+get forms with ajax 

I can't think of when we'd want more than one app on a page. 

> 'left', 'right' and 'center' where just ids, pluggable to any window manager. I might be wrong but if I understand this right then now we will have to load the panel layout in every app and hardcode the dom details each time. Preferably the app should not be responsible for the window management and/or hardcode targets e.g. 'body'. 

Right - you'll have to require the layout to use the three panel layout - it's important the page layout is de-coupled from the app. 

By 'hardocde the dom details', I think you mean not having to pass the `el: '#right' into the views? In backbone, it's very common for the functions that create views (like the app creates the page here) to link the html fragment that will be generated to some larger or encompassing dom (like the actual html page): http://backbonejs.org/#View-el

Requiring the panel is (at this point) needed to define the center and right panel classes. Defining the panel type in the app (and not the page) allows us to control behavior of panels, send them options, and re-use panel classes (such as a tool-panel class).

To me that's why decoupling here is even more important than usual: if we've decoupled the views properly, the components of the apps would be reusable *between* apps and will make the eventual implosion into one app easier. So, I favor decoupling over convenience here.

I'm not sure these are any more complex. I'll continue work to simplify the app structure as much as I can, tho.

> Other than this, this PR looks fantastic.

Thanks!